### DISCUSSION TITLE
关于Anime4K着色器问题

### DISCUSSION THREAD

**Discussion by sqq135** [hooke007/mpv_PlayKit#325] [mpv 通用]
想要配置合适自己的a4k着色器，但是对于各个着色器的应用场景不了解，看过https://hooke007.github.io/unofficial/mpv_shaders.html之后，要配置出相互搭配的着色器包仍力有未逮，在B站上https://www.bilibili.com/read/cv13643303看到另一教程，也是net-dw分支的来源之一，但是对于一个包里为什么这些东西不甚了解，为什么有的着色器包有Deblur或Denoise，有的没有，看2G一集和300m一集番剧用哪一个，都没有解释。
虽然这些工作可能应该先学习之后就会用，但是小白对于某些没法完全地学习，看番的画质问题又是很复杂的，想要在b站教程的基础上学习这些东西，希望大佬费心解惑。

**Comment by sqq135** [comment]
https://www.bilibili.com/read/cv13643303
https://hooke007.github.io/unofficial/mpv_shaders.html#id5

**Comment by sqq135** [comment]
这是DW分支中的染色器，就是不明白ctrl+7 和8、9、0之间差距过大的区别
```
CTRL+7     change-list glsl-shaders set "~~/shaders/Anime4K_Clamp_Highlights.glsl;~~/shaders/Anime4K_Restore_CNN_VL.glsl;~~/shaders/Anime4K_Upscale_CNN_x2_VL.glsl;~~/shaders/Anime4K_AutoDownscalePre_x2.glsl;~~/shaders/Anime4K_AutoDownscalePre_x4.glsl;~~/shaders/Anime4K_Upscale_CNN_x2_M.glsl" #menu: 着色器 > 预设组 > A4K-Mode-A (HQ)
_          change-list glsl-shaders set "~~/shaders/Anime4K_Clamp_Highlights.glsl;~~/shaders/Anime4K_Restore_CNN_M.glsl;~~/shaders/Anime4K_Upscale_CNN_x2_M.glsl;~~/shaders/Anime4K_AutoDownscalePre_x2.glsl;~~/shaders/Anime4K_AutoDownscalePre_x4.glsl;~~/shaders/Anime4K_Upscale_CNN_x2_S.glsl" #menu: 着色器 > 预设组 > A4K-Mode-A (Fast)
_          change-list glsl-shaders set "~~/shaders/Anime4K_Clamp_Highlights.glsl;~~/shaders/Anime4K_Restore_CNN_S.glsl;~~/shaders/Anime4K_Upscale_CNN_x2_S.glsl;~~/shaders/Anime4K_AutoDownscalePre_x2.glsl;~~/shaders/Anime4K_AutoDownscalePre_x4.glsl;~~/shaders/Anime4K_Upscale_CNN_x2_S.glsl" #menu: 着色器 > 预设组 > A4K-Mode-D (VeryFast)
CTRL+8     change-list glsl-shaders set "~~/shaders/Anime4K_Darken_HQ.glsl;~~/shaders/Anime4K_Thin_HQ.glsl;~~/shaders/Anime4K_Denoise_Bilateral_Mode.glsl;~~/shaders/Anime4K_Deblur_DoG.glsl" #menu: 着色器 > 预设组 > A4K-DTDD
CTRL+9     change-list glsl-shaders set "~~/shaders/SSimSuperRes.glsl;~~/shaders/SSimDownscaler.glsl" #menu: 着色器 > 预设组 > SSim-SR&DS
CTRL+0     change-list glsl-shaders set "~~/shaders/FSRCNNX_x2_8_0_4_1.glsl;~~/shaders/SSimDownscaler.glsl;~~/shaders/KrigBilateral.glsl" #menu: 着色器 > 预设组 > FSRCNNX+SSDS+Krig
```

**Reply by hooke007** [reply]
789都不是相同的着色器

**Comment by hooke007** [comment]
没有规定一定要用哪个，喜好＞原则。
对文字描述都无法理解的话，那就一个个试过去，去建立视觉上的效果观感，从而设立心理预期。（这个着色器应用之后会产生怎样的大致效果）

**Reply by sqq135** [reply]
主要看你介绍的着色器界面有这各种各样的效果，但是没有给建议的配置，B站上的教程只是给了答案，没有写原理注释。想要自己一个个试试，但发现可以多选，这就太复杂了。是想多看几个答案，互相参照学习下，再自己调整。
另外，暂停时调整也可以看到效果吗？
还有，之前lua执行失败（莫名其妙的，加了一行#注释原地址就失效了），能不能以一个简单的方式找原因，没有就算了

**Reply by hooke007** [reply]
> 但是没有给建议的配置

用途即建议。比如“去模糊”，那自然是模糊的才需要去模糊。

模糊产生的原因也可能是多重的，它可能是源本身就是blur的所以deblur就可以解决问题，它也可能是因为后处理scale放大产生的糊，那么这时可以用锐利的scale替换，但是这并不表示deblur不能解决糊的问题，你仍然可以deblur去解决的问题，当然和scale产生的效果肯定是有差距的（换句话讲解决问题的方法也不是唯一的）

当然我给出的最高建议依然是“喜好＞原则”，没有说一定要怎么怎么做的道理。

> 暂停时调整也可以看到效果吗

当然可以

> 之前lua执行失败

我看不懂这个描述
